### ISSUE TITLE
Upside down behaviour when calendar is close to the window edge

### ISSUE THREAD

**Issue by dbertella** [Hacker0x01/react-datepicker#1085]
![datepicker-weird-behaviour](https://user-images.githubusercontent.com/2563196/31891156-75401c12-b7fd-11e7-9d8e-f8bee2aa1136.gif)
Step to reproduce:
Open any of the examples and be sure the calendar opens close to the window edge.
Try to select any date, at click the calendar will go above the input and the user results in clicking the background. 
Once the calendar is above the text input you try to select a date and the calendar will go back to its position.
Basically it's impossible to select any date until you scroll or stretch the window

EDIT: The error occur in chrome on a mac and you have to find the right position, close enough to the bottom of the page to make it happening (the black line on the gif is my terminal underneath)
